### DISCUSSION TITLE
GPU TDP locked at 30W after replug in power

### DISCUSSION THREAD

**Discussion by TimCrownie** [BartoszCichecki/LenovoLegionToolkit#1101] [Q&A]
### Rules

- [X] I made myself familiar with the README.
- [X] I checked closed issues for similar ones and did not find a similar problem mentioned before.
- [X] I understand that, if insufficient information was provided, my issue will be closed without an answer.

### Version

2.18.0

### OS

Windows 10 22H2 19045.3803

### Device

Legion Y7000P 2023/Legion Slim 5i Gen 8

### BIOS version

M2CN29WW

### What's wrong?

Under normal circumstances, the upper limit of the TDP of the device's GPU should be 140W, and the minimum should remain at 65W; however, when the Legion Toolkit is started, when the power cord is unplugged and plugged back in, its TDP will be reduced to 30W and be Forcibly locked at this value, switching performance mode cannot unlock it.
However, it can be solved by exit Legion Toolkit, switch performance mode and start up Legion Tookit again.

### How to reproduce the bug?

1. Start Legion Toolkit, and switch performance mode to customize mode
2. Unplug the power cable and replug in
3. Fire up something(like Furmark) to verify if GPU's TDP is locked at 30W again
4. Well... it is. As you can see, the power draw is locked at 30W max due to "Power Limit". But according to NVIDIA, this GPU's lowest TDP is 35W. And according to my personal test on this device, it'll do a lowest TDP at 65W on customized mode with all enhance options turned off, locked under 60W(but unstable) at quiet mode, 80W at balance mode, 125W at rampage(or turbo?) mode, and hit max 140W TDP at customized mode, with all enhance options enabled and cranked to max value.
<img width="382" alt="屏幕截图 2024-01-01 001751" src="https://github.com/BartoszCichecki/LenovoLegionToolkit/assets/63519560/8e89a21d-be42-4644-925f-a341a6f68404">

### What is the behavior that you expected?

Well, it's a easy issue to fix manually, but that's also really annoying.
So...is there anybody could help me fix this bug which always made my RTX 4060 Laptop behaves even worse than a GTX 1050 Max-Q?

### Logs

[log_2023_12_31_16_16_20.txt](https://github.com/BartoszCichecki/LenovoLegionToolkit/files/13801675/log_2023_12_31_16_16_20.txt)

### Additional information

I'm not sure if it's an issue that only happens on this(my) particular device, so maybe you can't reproduce this bug on your device?
And by the way, I also have this bug on Legion Zone.

**Comment by BartoszCichecki** [comment]
LLT doesn't control power limits in power modes other than custom. It is entirely up to BIOS/EC to control these. I will convert this issue into a discussion in case someone can help, because this isn't something I can fix.

However, maybe you can work around this issue with Actions? Set up an action with multiple triggers for "Power Mode = Performance" *and* "AC adapter plugged in" and add steps:
1. Power Mode = Balance
2. Delay = 1 second
3. Power Mode = Performance.
